packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+[A-Za-z]+                                 return 'ALPHA';
+":"                                       return 'COLON';
+\+                                        return 'PLUS';
+","                                       return 'COMMA';
+"="                                       return 'EQUALS';
+\=                                        return 'EQUALS';
+"*"                                       return 'MULT';
+\#                                        return 'BRKT';
+[\_]                                      return 'UNDERSCORE';
+"."                                       return 'DOT';
+"&"                                       return 'AMP';
+\-                                        return 'MINUS';
+[0-9]+                                    return 'NUM';
+\s                                        /* skip */
+";"                                       return 'SEMI';
+[!"#$%&'*+,-.`?\\_/]                      return 'PUNCTUATION';
+<<EOF>>                                   return 'EOF';
+
+/lex
+
+%start start
+
+%% /* language grammar */
+
+start
+  : eol start
+  | directive start
+  | XYCHART chartConfig start
+	| XYCHART start
+  | document
+	;
+
+chartConfig
+  : CHART_ORIENTATION                     { yy.setOrientation($1); }
+  ;
+
+document
+	: /* empty */
+	| document statement
+	;
+
+line
+	: statement
+	;
+
+statement
+  : statement eol
+  | title title_value                     { yy.setDiagramTitle($title_value.trim()); }
+  | X_AXIS parseXAxis
+  | Y_AXIS parseYAxis
+  | LINE parseLineData                    { yy.setLineData({text: '', type: 'text'}, $parseLineData); }
+  | LINE text parseLineData               { yy.setLineData($text, $parseLineData); }
+  | BAR parseBarData                      { yy.setBarData({text: '', type: 'text'}, $parseBarData); }
+  | BAR text parseBarData                 { yy.setBarData($text, $parseBarData); }
+	;
+
+parseLineData
+  : SQUARE_BRACES_START NUMBER_WITH_DECIMAL parseLineData   {$parseLineData.unshift(Number($NUMBER_WITH_DECIMAL)); $$ = $parseLineData}
+  | COMMA NUMBER_WITH_DECIMAL parseLineData                 {$parseLineData.unshift(Number($NUMBER_WITH_DECIMAL)); $$ = $parseLineData;}
+  | SQUARE_BRACES_END                                       {$$ = []}
+  ;
+
+parseBarData
+  : SQUARE_BRACES_START NUMBER_WITH_DECIMAL parseBarData    {$parseBarData.unshift(Number($NUMBER_WITH_DECIMAL)); $$ = $parseBarData}
+  | COMMA NUMBER_WITH_DECIMAL parseBarData                  {$parseBarData.unshift(Number($NUMBER_WITH_DECIMAL)); $$ = $parseBarData;}
+  | SQUARE_BRACES_END                                       {$$ = []}
+  ;
+
+parseXAxis
+  : text                                                          {yy.setXAxisTitle($text);}
+  | text xAxisBandData                                            {yy.setXAxisTitle($text); yy.setXAxisBand($xAxisBandData);}
+  | text NUMBER_WITH_DECIMAL ARROW_DELIMITER NUMBER_WITH_DECIMAL  {yy.setXAxisTitle($text); yy.setXAxisRangeData(Number($2), Number($4));}
```

Comment:
You can give them custom aliases should you need to:
```
text NUMBER_WITH_DECIMAL\[first] ARROW_DELIMITER NUMBER_WITH_DECIMAL\[second]
{yy.setXAxisTitle($text); yy.setXAxisRangeData(Number($first), Number($second));}
```
But they are also [auto-numbered](https://gerhobbelt.github.io/jison/docs/#auto-numbered-named-accessors):
```
{yy.setXAxisTitle($text); yy.setXAxisRangeData(Number($NUMBER_WITH_DECIMAL1), Number($$NUMBER_WITH_DECIMAL2));}
```

Reply:
Thanks for the suggestion, Used the auto-numbered one